pkg/storage/inmemory/inmemory.go

```diff
@@ -122,12 +129,24 @@ func (db *DB) List(_ context.Context, sinceVersion string) ([]*databroker.Record
 
 // Put replaces or inserts a record in the db.
 func (db *DB) Put(_ context.Context, id string, data *anypb.Any) error {
+	defer db.onchange.Broadcast()
 	db.replaceOrInsert(id, func(record *databroker.Record) {
 		record.Data = data
 	})
 	return nil
 }
 
+// Sync returns the underlying signal.Signal binding channel to the caller.
+// Then the caller can listen to the channel for detecting changes.
+func (db *DB) Sync(ctx context.Context) chan struct{} {
+	go func() {
+		defer db.onchange.Unbind(db.syncCh)
```

Comment:
why do we unbind on ctx.Done? If I used a DB across multiple connections wouldn't this result in all signals being stopped?

Reply:
Ah right.